src/main/java/uk/gov/hmcts/divorce/divorcecase/model/Solicitor.java

```diff
@@ -94,4 +104,52 @@ public boolean hasOrgId() {
         }
         return false;
     }
+
+    public String getAddress() {
+        if ((YesOrNo.NO == getAddressOverseas() || null == getAddressOverseas()) && (this.address != null && !this.address.equals(""))) {
+            String ukPostcodeRegex = "([A-Za-z][A-Ha-hJ-Yj-y]?[0-9][A-Za-z0-9]? ?[0-9][A-Za-z]{2}|[Gg][Ii][Rr] ?0[Aa]{2})";
+            Matcher matcher = Pattern.compile(ukPostcodeRegex).matcher(this.address);
+            if (matcher.find()) {
+                String postcode =  matcher.group(1);
+                String addressPostcodeStripped = this.address.replace(postcode, "")
+                    .replace("\n\n", "\n")
+                    .replace(",,", ",")
+                    .replace("\n,\n", "\n")
+                    .replace(", , ", ", ")
+                    .replace(" \n, \n", " \n")
+                    .replace("\n ,\n", "\n")
+                    .replace(", \n ,", ",")
+                    .replace(" \n  \n", " \n")
+                    .replace("\n \n ", "\n ")
+                    .replace(" \n \n", " \n")
```

Comment:
Maybe we could use a couple of regular expressions to simplify this matching? Something like this (probably will need to be tweaked):
```suggestion
                    .replaceAll("\n[, ]*\n", "\n")
                    .replaceAll(",[\n ]*,", ",")
```

Reply:
No longer necessary - but worked first time in my test scenarios!  nice one :)  I think you just became our regex goto ;)